lib/cooked_post_processor.rb

```diff
@@ -373,6 +372,38 @@ def each_responsive_ratio
       .each { |r| yield r if r > 1 }
   end
 
+  def optimize_image!(img, upload, cropped: false)
+    w, h = img["width"].to_i, img["height"].to_i
+
+    thumbnail = upload.thumbnail(w, h)
+    if thumbnail && thumbnail.filesize.to_i < upload.filesize
+      img["src"] = thumbnail.url
+
+      srcset = +""
+
+      each_responsive_ratio do |ratio|
+        resized_w = (w * ratio).to_i
+        resized_h = (h * ratio).to_i
+
+        if !cropped && upload.width && resized_w > upload.width
+          cooked_url = UrlHelper.cook_url(upload.url, secure: upload.secure?)
```

Comment:
feels like the two repeated lines here, could be extracted in a method

Reply:
Not sure we need a whole method for that. Just re-order the logic a tiny bit?

```ruby

upload_url = if !cropped && upload.width && resized_w > upload.width
  upload.url
elseif t = upload.thumbnail(resized_w, resized_h)
  t.url
end

if upload_url.present?
  cooked_url = UrlHelper.cook_url(upload_url, secure: upload.secure?)
  srcset << ", #{cooked_url} #{ratio.to_s.sub(/\.0$/, "")}x"
end